common/include/pcl/impl/point_types.hpp

```diff
@@ -245,6 +247,19 @@ namespace pcl
     }; \
   };
 
+#define PCL_ADD_UNION_POINT3D \
+  union EIGEN_ALIGN16 { \
+    float data[3]; \
```

Comment:
Why did you make this 3 values long, not just 2?

Reply:
I saw the XYZ implementation had data as 4 values long. So I just followed a pattern. I can step it down to 2 values.

Reply:
The XYZ implementation is 4 long instead of 3 because 4 floats align nicely, and they have a function to get a Vector4f map (useful for some computations). Right now I don't see similar benefits to make the XY implementation 3 long instead of 2

Reply:
Okay got it

Reply:
I have made the changes.